mobile/src/main/java/ch/epfl/sdp/mobile/state/game/core/PuzzleGameDelegate.kt

```diff
@@ -0,0 +1,101 @@
+package ch.epfl.sdp.mobile.state.game.core
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import ch.epfl.sdp.mobile.application.authentication.AuthenticatedUser
+import ch.epfl.sdp.mobile.application.chess.Puzzle
+import ch.epfl.sdp.mobile.application.chess.baseGame
+import ch.epfl.sdp.mobile.application.chess.engine.Game
+import ch.epfl.sdp.mobile.application.chess.engine.NextStep
+import ch.epfl.sdp.mobile.application.chess.engine.rules.Action
+import ch.epfl.sdp.mobile.state.game.delegating.DelegatingPuzzleInfoState
+import ch.epfl.sdp.mobile.ui.puzzles.PuzzleInfoState
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+/**
+ * An implementation of [MutableGameDelegate] which implements [Puzzle] mechanics.
+ *
+ * @param user the [State] of the current [AuthenticatedUser].
+ * @param puzzle the underlying [Puzzle].
+ * @param delegate the [DelegatingPuzzleInfoState] that provides the puzzle's state info.
+ * @param scope the [CoroutineScope] used to read and write the match updates.
+ */
+class PuzzleGameDelegate(
+    private val user: State<AuthenticatedUser>,
+    private val puzzle: Puzzle,
+    private val delegate: DelegatingPuzzleInfoState,
+    private val scope: CoroutineScope,
+) : MutableGameDelegate, PuzzleInfoState by delegate {
+
+  /** The underlying snapshot-aware [Game]. */
+  private var backing by mutableStateOf(puzzle.baseGame())
+
+  override var game: Game
+    get() = backing
+    set(value) {
+      backing = value // Local updates
+    }
```

Comment:
Could this be simplified to :

```kotlin
override var game by mutableStateOf(puzzle.baseGame())
```

?

Reply:
Yes. Fixed in https://github.com/epfl-SDP/android/pull/309/commits/1adab13015e5b5a350cb4f11705b788704bd94ec